### PR TITLE
Add animations and indicator while ON

### PR DESCRIPTION
**Pull request by eLtMosen**
- Add indicator to active flashlight to not mistake it for a glitch.
- Animate the added flashCircle to transform into the active flashlight
- Transform to either a square or stay in circle shape depending on watch screen.

On square display
https://user-images.githubusercontent.com/15074193/153714139-874e0642-b9c1-43f0-a831-35c205c15ff3.mp4

On round display
https://user-images.githubusercontent.com/15074193/153714147-f3792b09-aa59-4c87-85eb-9a261f2f8550.mp4

### CODE REVIEW

File: src/main.qml [hunk 1]
```diff
@@ -17,45 +18,46 @@
...
             }
+            width: flashCircle.width * .7
+            height: width
+            color: flashCircle.flashOn ? "#F0F0F0" : "#FFF"
+            name:  flashCircle.flashOn ? "ios-bulb-outline" : "ios-bulb"
         }
+
+        MouseArea {
+            anchors.fill: flashCircle
+            onClicked: flashCircle.flashOn ? flashCircle.flashOn = false : flashCircle.flashOn = true
```
Comment (FlorentRevest) [inline src/main.qml:54]: `flashCircle.flashOn = !flashCircle.flashOn` is usually how we do this but no big deal, this is also okay

File: src/main.qml [hunk 2]
```diff
@@ -17,45 +18,46 @@
...
-                onOffAnimation.to = 0
-                onOffAnimation.start()
-            } else {
-                onOffAnimation.from = 0
-                onOffAnimation.to = 1
-                onOffAnimation.start()
+        Icon {
+            anchors {
+                centerIn: flashCircle
+                verticalCenterOffset: Dims.h(1)
```
Comment (FlorentRevest) [inline src/main.qml:44]: Dims.h is a percentage of the screen size so the offset will be the same whether the icon is large (flashOn) or not. This feels a bit wrong
If this is just 1% of the overall screen and you didn't notice the discrepancy, I'm tempted to say this is not necessary... :p 
And if the icon is not well aligned I think it's best to fix the icon than the app with this kind of hard-coded offsets.

**Review by FlorentRevest — APPROVED** [review]
Looks really good!
Just two nits but these are not important so this is approved for merging

File: src/main.qml (hunk 2 above)
Comment (eLtMosen) [reply src/main.qml:44]: Well, it is intentional to have a proportionally larger offset when the icon is small and proportionally small offset when the icon is large :P
But now that you are uncomfortable with how this is implemented, i'll pick the clean option to center the icons for the usage in narrow spaced circles (!flashOn) and ignore the change for flashOn.
Original intention was to offset the icon 1% when small, and "don't care" about the offset when enlarged/flashOn.
Which is also reached by fixing the icon scg.
Reverting the offset here so we can merge.
Fixing the icons in another PR.

Comment (FlorentRevest) [reply src/main.qml]: > Well, it is intentional to have a proportionally larger offset

Yes, but notice that Dims.h is proportional to... the screen size. Which is constant. So 

> when the icon is small and proportionally small offset when the icon is large :P

So this is not happening in practice. The offset is always the same: 1% of your screen height

**Pull request merged**